Src/HALALMock/Services/Communication/Ethernet/TCP/Socket.cpp

```diff
@@ -174,8 +198,8 @@ void Socket::reset(){
 
 void Socket::send(){
 	while (!tx_packet_buffer.empty()) {
-        HeapPacket packet = tx_packet_buffer.front();
-        ssize_t sent_bytes = ::send(socket_fd, packet.build(), packet.get_size(), 0);
+        HeapPacket *packet = tx_packet_buffer.front();
```

Comment:
Packet*

Reply:
can you explain this, at the end packet is a virtual class to store data shouldn't I use one class derivated from packet